cfme/tests/cli/test_appliance_console_db_restore.py

```diff
@@ -0,0 +1,471 @@
+    nfs_smb = SSHClient(**connect_kwargs)
+    dump_filename = "/tmp/db_dump_{}".format(fauxfactory.gen_alphanumeric())
+    appl1.ssh_client.get_file("/tmp/evm_db.backup", dump_filename)
+    nfs_smb.put_file(dump_filename, "{}share.backup".format(loc))
+
+
+@pytest.mark.tier(2)
+@pytest.mark.uncollectif(
+    lambda: not store.current_appliance.is_downstream)
+def test_appliance_console_restore_db_local(request, soft_assert, get_appliances_with_providers):
+    """ Test single appliance backup and restore, configures appliance with providers,
+    backs up database, restores it to fresh appliance and checks for matching providers.
+    """
+    appl1, appl2 = get_appliances_with_providers
+    # Transfer v2_key and db backup from first appliance to second appliance
+    fetch_v2key(appl1, appl2)
+    fetch_db_local(appl1, appl2)
+    # Restore DB on the second appliance
+    appl2.evmserverd.stop()
+    appl2.db.drop()
+    command_set = ('ap', '', '4', '1', '', TimedCommand('y', 60), '')
+    appl2.appliance_console.run_commands(command_set)
+    appl2.start_evm_service()
+    appl2.wait_for_web_ui()
+    # Assert providers on the second appliance
+    assert set(appl2.managed_provider_names) == set(appl1.managed_provider_names), (
+        'Restored DB is missing some providers'
+    )
+    # Verify that existing provider can detect new VMs on the second appliance
+    virtual_crud = provider_app_crud(VMwareProvider, appl2)
+    vm = provision_vm(request, virtual_crud)
+    soft_assert(vm.provider.mgmt.is_vm_running(vm.name), "vm running")
+
+
+@pytest.mark.tier(2)
+@pytest.mark.uncollectif(
+    lambda: not store.current_appliance.is_downstream or store.current_appliance.version < '5.9')
+def test_appliance_console_restore_pg_basebackup_ansible(
+        request, soft_assert, get_appliances_with_ansible):
+    appl1 = get_appliances_with_ansible
+    providers_before_restore = set(appl1.managed_provider_names)
+    # Restore DB on the second appliance
+    appl1.evmserverd.stop()
+    command_set = ('ap', '', '4', '1', '/tmp/backup/base.tar.gz', TimedCommand('y', 60), '')
+    appl1.appliance_console.run_commands(command_set)
+    appl1.start_evm_service()
+    appl1.wait_for_web_ui()
+    # Assert providers on the second appliance
+    assert providers_before_restore == set(appl1.managed_provider_names), (
+        'Restored DB is missing some providers'
+    )
+    # Verify that existing provider can detect new VMs on the second appliance
+    virtual_crud = provider_app_crud(VMwareProvider, appl1)
+    vm = provision_vm(request, virtual_crud)
+    soft_assert(vm.provider.mgmt.is_vm_running(vm.name), "vm running")
+    # bug in ansible stops it working first time after restore, requires reboot
+    appl1.reboot()
+    assert wait_for(func=lambda: appl1.is_embedded_ansible_running, num_sec=30)
+    assert wait_for(func=lambda: appl1.is_rabbitmq_running, num_sec=30)
+    assert wait_for(func=lambda: appl1.is_nginx_running, num_sec=30)
```

Comment:
**REQUIRED**
I don't assertion `wait_for` call makes sense. It will raise `TimedOutError`.